src/main/java/axiomq/com/starwars/config/JwtFilter.java

```diff
@@ -0,0 +1,48 @@
+package axiomq.com.starwars.config;
+
+import axiomq.com.starwars.services.implementations.CustomUserDetailsService;
+import axiomq.com.starwars.util.JwtUtil;
+import lombok.AllArgsConstructor;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@Component
+@AllArgsConstructor
+public class JwtFilter extends OncePerRequestFilter {
+
+    private JwtUtil jwtUtil;
+
+    private CustomUserDetailsService userDetailsService;
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain) throws ServletException, IOException {
+        String authorizationHeader = request.getHeader("Authorization");
+        String token = null;
+        String username = null;
+        if(authorizationHeader != null && authorizationHeader.startsWith("Bearer ")){
+            token = authorizationHeader.substring(7);
+            username = jwtUtil.extractUsername(token);
+        }
+        if(username != null && SecurityContextHolder.getContext().getAuthentication() == null){
```

Comment:
unnecessary block `SecurityContextHolder.getContext().getAuthentication() == null` is always true